### PR TITLE
feat: allow swap per nodepool for both control plane and agent

### PR DESCRIPTION
**Pull request by Silvest89**
Just trying to step on the gas for some PR's ;]

Won't be using swap myself but as some people love this feature.

As a reference as well https://kubernetes.io/blog/2023/08/24/swap-linux-beta/
Kubernetes 1.28 has beta support for swap :)

Original PR:
https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/pull/803

### CODE REVIEW

**Comment by mysticaltech** [comment]
@Silvest89 Thanks, LGTM. Please see the two other pending but related PRs.

**Comment by Silvest89** [comment]
@mysticaltech which two?

**Comment by mysticaltech** [comment]
Sorry @Silvest89, was on mobile yesterday. Here's you go:
https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/pull/803
https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/pull/803

I guess the zRam is kind of related in the spirit of more memory more or less.

**Comment by mysticaltech** [comment]
@Silvest89 Thanks for this, awesome! Review coming A-S-A-P.

File: kube.tf.example [hunk 1]
```diff
@@ -108,6 +108,7 @@ module "kube-hetzner" {
       labels      = [],
       taints      = [],
       count       = 1
+      swap_size   = "2G" # remember to add the suffix, examples: 512M, 1G
```
Comment (mysticaltech) [inline kube.tf.example:111]: Let's comment it out by default. so that swaping stays optional.

**Review by mysticaltech — APPROVED** [review]
LGTM. Will just comment out the line in kube.tf.example myself. Good job!

**Pull request merged**